tests/php/test-ip-addresses.php

```diff
@@ -71,4 +71,93 @@ public function test_get_client_ip_address() {
 			unset( $_SERVER[ $header ] );
 		}
 	}
+
+	public function test_rsa_trusted_proxies() {
+		$rsa = Restricted_Site_Access::get_instance();
+
+		$_SERVER['REMOTE_ADDR'] = '127.0.0.1';
+
+		// Test that if the REMOTE_ADDR matches our proxy, we return a proper IP.
+		add_filter( 'rsa_trusted_proxies', function() {
+			return array( '127.0.0.1/24' );
+		} );
+
+		$this->assertSame( '127.0.0.1', $rsa::get_client_ip_address() );
+
+		// Test that if the REMOTE_ADDR doesn't match our proxy, we return an empty string.
+		add_filter( 'rsa_trusted_proxies', function() {
+			return array( '10.0.0.0/8' );
+		} );
+
+		$this->assertSame( '', $rsa::get_client_ip_address() );
+
+		// Test if we have multiple proxies and one matches, we return a proper IP.
+		add_filter( 'rsa_trusted_proxies', function() {
+			return array( '10.0.0.0/8', '127.0.0.1' );
+		} );
+
+		$this->assertSame( '127.0.0.1', $rsa::get_client_ip_address() );
+
+		// Reset the filter.
+		add_filter( 'rsa_trusted_proxies', '__return_empty_array' );
```

Comment:
```suggestion
```

Redundant, `WP_UnitTestCase` resets the filters between each test. ([source ref](https://github.com/WordPress/wordpress-develop/blob/13e076e6a48deb388bcba3a23beea2a2465a82d5/tests/phpunit/includes/abstract-testcase.php#L194))